kernel/names.ml

```diff
@@ -1104,3 +1106,28 @@ type lname = Name.t CAst.t
 type lstring = string CAst.t
 
 let lident_eq = CAst.eq Id.equal
+
+(** Evaluable references (whose transparency can be controlled) *)
+
+module Evaluable = struct
```

Comment:
I'd rather have this type defined outside of the kernel, and rather adapt the semi-internal tableKey in CClosure accordingly.

Reply:
No, I meant that I'd rather have a separation of kernel-facing types and high-level ones. That it, keep Tacred.evaluable_reference in the upper layers and tweak tableKey or introduce a new type in the kernel. Otherwise we have a new coupling between tactic-related code and kernel APIs, which, although not critical, could be a bit annoying for future API changes.

Reply:
(I am also fine with merging this as-is, though.)

Reply:
OK, I have a version where I basically duplicate the type, with a function to convert to the kernel version, but it probably requires some bikeshedding (as well as extra overlay work). So let's do this in a second step then?